### PR TITLE
Code3

### PR DESCRIPTION
**Pull request by arb**
Updated code version
Updated to use ES6 class
Cleaned up documentation to discourage `var` use

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -58,41 +44,49 @@ internals.compile = function (override) {
...
-internals.Mimos.prototype.path = function (path) {
+class Mimos {
+    constructor(options) {
 
-    const extension = Path.extname(path).slice(1).toLowerCase();
-    const mime = this._byExtension[extension] || {};
-
-    if (mime.predicate) {
-        return mime.predicate(Hoek.clone(mime));
+        options = options || {};
```
Comment (cjihrig) [inline lib/index.js]: You could `Object.assign()` this.

File: lib/index.js [hunk 2]
```diff
@@ -58,41 +44,49 @@ internals.compile = function (override) {
...
+}
 
-    if (mime.predicate) {
-        return mime.predicate(Hoek.clone(mime));
-    }
+internals.base = internals.compile();       // Prevents an expensive copy on each constructor when no customization is needed
 
-    return mime;
-};
+module.exports = Mimos;
```
Comment (geek) [inline lib/index.js]: Move to the top where class is created

File: lib/index.js [hunk 3]
```diff
@@ -58,41 +44,49 @@ internals.compile = function (override) {
...
+        this._byExtension = result.byExtension;
     }
+    path(path) {
 
-    return mime;
-};
+        const extension = Path.extname(path).slice(1).toLowerCase();
+        const mime = this._byExtension[extension] || {};
 
+        if (mime.predicate) {
```
Comment (geek) [inline lib/index.js]: could check its a function

**Comment by cjihrig** [comment]
LGTM

**Pull request merged**